iohub/ngff/utils.py

```diff
@@ -0,0 +1,485 @@
+        Maximum chunk size in bytes. Defaults to 500e6 (500MB).
+
+    Examples
+    --------
+    Create a new plate with positions and channels:
+    create_empty_plate(
+        store_path=Path("/path/to/store"),
+        position_keys=[("A", "1", "0"), ("A", "1", "1")],
+        channel_names=["DAPI", "FITC"],
+        shape=(1, 1, 256, 256, 256)
+    )
+
+    Create a plate with custom chunk size and scale:
+    create_empty_plate(
+        store_path=Path("/path/to/store"),
+        position_keys=[("A", "1", "0")],
+        channel_names=["DAPI"],
+        shape=(1, 1, 256, 256, 256),
+        chunks=(1, 1, 128, 128, 128),
+        scale=(1, 1, 0.5, 0.5, 0.5)
+    )
+
+    Notes
+    -----
+    - If `chunks` is not provided, the function calculates an appropriate
+    chunk size to keep the chunks under the specified `max_chunk_size_bytes`.
+    - The function ensures that positions and channels are appended to an
+    existing plate if they are not already present.
+    """
+
+    # Limiting the chunking to 500MB
+    if chunks is None:
+        chunk_zyx_shape = _calculate_zyx_chunk_size(
+            shape,
+            np.dtype(dtype).itemsize,
+            max_chunk_size_bytes,
+        )
+
+        chunks = 2 * (1,) + chunk_zyx_shape
+
+    # Create plate
+    output_plate = open_ome_zarr(
+        str(store_path), layout="hcs", mode="a", channel_names=channel_names
+    )
+
+    # Create positions
+    for position_key in position_keys:
+        position_key_string = "/".join(position_key)
+        # Check if position is already in the store, if not create it
+        if position_key_string not in output_plate.zgroup:
+            position = output_plate.create_position(*position_key)
+            _ = position.create_zeros(
+                name="0",
+                shape=shape,
+                chunks=chunks,
+                dtype=dtype,
+                transform=[TransformationMeta(type="scale", scale=scale)],
+            )
+        else:
+            position = output_plate[position_key_string]
+
+    # Check if channel_names are already in the store, if not append them
+    for channel_name in channel_names:
+        metadata_channel_names = position.channel_names
+        if channel_name not in metadata_channel_names:
+            position.append_channel(channel_name, resize_arrays=True)
+
+
+def apply_transform_to_zyx_and_save(
+    func: Callable,
+    position_key: str,
+    input_store_path: Path,
+    output_store_path: Path,
+    input_channel_indices: Union[list[slice], list[list[int]]],
+    output_channel_indices: Union[list[slice], list[list[int]]],
+    input_time_indices: int,
+    output_time_indices: int,
+    **kwargs,
+) -> None:
+    """
+    Load a CZYX array from a Position object,
+    apply a transformation, and save the result.
+
+    Parameters
+    ----------
+    func : Callable
+        The function to be applied to the data.
+        Should take a CZYX array and return a transformed CZYX array.
```

Comment:
This description can be more explicitly typed as `Callable[[ArrayLike], ArrayLike]`. However reading the code it seems like `Callable[[NDArray, ...], NDArray]` is the actual behavior.

Reply:
what was this for?

Reply:
We don't enforce type checking in iohub. But it's a nice to have for downstream developers. For example you get better LSP semantics.

Reply:
Just improved the typing and docs. Thanks @ziw-liu.